qa/rpc-tests/wallet_orchard.py

```diff
@@ -0,0 +1,145 @@
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    NU5_BRANCH_ID,
+    assert_equal,
+    get_coinbase_address,
+    nuparams,
+    start_nodes,
+    wait_and_assert_operationid_status
+)
+
+from decimal import Decimal
+
+# Test wallet behaviour with the Orchard protocol
+class WalletOrchardTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 4
+
+    def setup_nodes(self):
+        return start_nodes(self.num_nodes, self.options.tmpdir, [[
+            '-experimentalfeatures',
+            '-orchardwallet',
+            nuparams(NU5_BRANCH_ID, 210),
+        ]] * self.num_nodes)
+
+    def run_test(self):
+        # Sanity-check the test harness
+        assert_equal(self.nodes[0].getblockcount(), 200)
+
+        # Get a new orchard-only unified address
+        acct1 = self.nodes[1].z_getnewaccount()['account']
+        addrRes1 = self.nodes[1].z_getaddressforaccount(acct1, ['orchard'])
+        assert_equal(acct1, addrRes1['account'])
+        assert_equal(addrRes1['pools'], ['orchard'])
+        ua1 = addrRes1['unifiedaddress']
+
+        # Verify that we have only an Orchard component
+        receiver_types = self.nodes[0].z_listunifiedreceivers(ua1)
+        assert_equal(set(['orchard']), set(receiver_types))
+
+        # Verify balance
+        assert_equal({'pools': {}, 'minimum_confirmations': 1}, self.nodes[1].z_getbalanceforaccount(acct1))
+
+        # Send some sapling funds to node 2 for later spending after we split the network
+        acct2 = self.nodes[2].z_getnewaccount()['account']
+        addrRes2 = self.nodes[2].z_getaddressforaccount(acct2, ['sapling', 'orchard'])
+        assert_equal(acct2, addrRes2['account'])
+        ua2 = addrRes2['unifiedaddress']
+        saplingAddr2 = self.nodes[2].z_listunifiedreceivers(ua2)['sapling']
+
+        recipients = [{"address": saplingAddr2, "amount": Decimal('10')}]
+        myopid = self.nodes[0].z_sendmany(get_coinbase_address(self.nodes[0]), recipients, 1, 0)
+        wait_and_assert_operationid_status(self.nodes[0], myopid)
+
+        # Mine the tx & activate NU5
+        self.sync_all()
+        self.nodes[0].generate(10)
+        self.sync_all()
+
+        # Check the value sent to saplingAddr2 was received in node 2's account
+        assert_equal(
+                {'pools': {'sapling': {'valueZat': Decimal('1000000000')}}, 'minimum_confirmations': 1}, 
+                self.nodes[2].z_getbalanceforaccount(acct2))
+
+        # Node 0 shields some funds
+        # t-coinbase -> Orchard
+        recipients = [{"address": ua1, "amount": Decimal('10')}]
+        myopid = self.nodes[0].z_sendmany(get_coinbase_address(self.nodes[0]), recipients, 1, 0)
+        wait_and_assert_operationid_status(self.nodes[0], myopid)
+
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
```

Comment:
There are helper methods defined like this in `wallet_changeindicator.py`:
```
    def generate_and_sync(self):
        self.sync_all()
        self.nodes[0].generate(1)
        self.sync_all()
```
and like this in `wallet_listreceived.py`:
```
    def generate_and_sync(self, new_height):
        current_height = self.nodes[0].getblockcount()
        assert(new_height > current_height)
        self.sync_all()
        self.nodes[0].generate(new_height - current_height)
        self.sync_all()
        assert_equal(new_height, self.nodes[0].getblockcount())
```

Consider moving them to be a utility method on a mininode and using that here.